htslib/sam.h

```diff
@@ -1090,6 +1090,73 @@ hts_pos_t bam_cigar2rlen(int n_cigar, const uint32_t *cigar);
 HTSLIB_EXPORT
 hts_pos_t bam_endpos(const bam1_t *b);
 
+/*! @typedef
+ *  @abstract Flag filtering settings structure
+ */
+typedef struct sam_flag_filter_s {
+    uint32_t has_all:1, has_none:1, has_any:1, has_omit_any:1, has_exact:1, :27;
+    // To pass this filter, a read must have...
+    unsigned int all;       // all of these flags
+    unsigned int none;      // none of these
+    unsigned int any;       // at least one of these
+    unsigned int omit_any;  // OMIT at least one of these
```

Comment:
I've been strenuously avoiding `not_all` because it is ambiguous between “it must be that not all of these are 1“ and “it must be that all of these are not 1” — which are very different.

Reply:
Note {any,all}-{set,unset} is what we ended up picking in bcftools too, albeit that was "skip" rather than "keep", as it was being used as a filter.